### PR TITLE
Mongo db team

### PR DESCRIPTION
**Pull request by Shira-Abrams**
fix all merge

### CODE REVIEW

**Review by moriya2004 — APPROVED** [review]

**Review by BatshevaTavger — APPROVED** [review]

File: server-side/controllers/user.controller.js [hunk 1]
```diff
@@ -1,114 +1,87 @@
...
-      name,
-      password: hashedPassword,
-      email,
-    });
+    
+    if (req.file ) 
+     req.body.profileImage=req.file.originalname;
+     req.body.password= await bcrypt.hash(password, 10);
+    const newUser = new Users(req.body);
+    console.log('newUser',newUser);
```
Comment (SariGlick) [inline server-side/controllers/user.controller.js:41]: Is it necessary? remove all unnecessary logs

File: server-side/controllers/visitedWebSite.controller.js [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
-            const visitedWebsites = await visitedWebsites.find();
-            res.json(visitedWebsites);
-        } catch (err) {
-            res.status(500).json({ message: err.message });
+export const getAllVisitedWebsites = async (req, res,next) => {
+    try {
+        const visitedWebsites = await VisitedWebsite.find().populate('websiteId').select('-__v');
+        res.json(visitedWebsites);
+    } catch (err) {
+        next({message:err.message})
```
Comment (SariGlick) [inline server-side/controllers/visitedWebSite.controller.js:8]: always send with suitable status code
also add return after the call to next
fix all occurrences in file

File: server-side/controllers/visitedWebSite.controller.js [hunk 3]
```diff
@@ -1,53 +1,52 @@
...
-        } catch (err) {
-            res.status(400).json({ message: err.message });
+        res.json(visitedWebsite);
+    } catch (err) {
+        next({message:err.message})
+    }
+};
+export const updateVisitedWebsite = async (req, res,next) => {
+    try {
+        const updatedVisitedWebsite = await VisitedWebsite.findByIdAndUpdate(req.params.id, req.body, { new: true });
```
Comment (SariGlick) [inline server-side/controllers/visitedWebSite.controller.js:33]: add validation for parameters wherever you use them

**Review by SariGlick — CHANGES_REQUESTED** [review]

File: server-side/controllers/preference.controller.js [hunk 4]
```diff
@@ -79,28 +6,27 @@ export const getAllPreference = async (req, res, next) => {
         const allPreferences = await Preference.find().select('-__v');
         return res.send(allPreferences);
     } catch (error) {
-        return next({ message: error.message })
+         next({ message: error.message })
```
[the remark this answers is not in the document]
Comment (moriya2004) [reply server-side/controllers/preference.controller.js:9]: we have mw that send the status

**Review by moriya2004 — APPROVED** [review]

File: client-side/src/components/settings/manageNotifications/EmailFrequency.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  'never':'🚫',
+  'daily':'📅',
+  'weekly':'🗓️',
+  'monthly':'📆',
+  'yearly':'📅'
+};
+
+
+const EmailFrequency = () => {
+  const [emailFrequency, setEmailFrequency] = useState('');
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/EmailFrequency.jsx:16]: why does emailFrequency needs to be in state? on user clicks on save buuton, you get the value from the select and save it

File: client-side/src/components/settings/manageNotifications/EmailFrequency.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  const [message, setMessage] = useState('');
+  const preferenceId = '66930c2e2aad987e24078e12';//TO DO: get from state
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/EmailFrequency.jsx:18]: I dont see you fixed it, and now I realize that it's also unnecessary in state, you get it from props with all the preferences data.

File: client-side/src/components/settings/manageNotifications/EmailFrequency.jsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      setMessage('Email frequency preference updated successfully!');
+    } catch (error) {
+      console.error('Error updating email frequency preference:', error);
+      setMessage('Error updating email frequency preference. Please try again later.');
+    }
+  };
+  const handleChange = (e) => {
+    const selectedFrequency = e.target.value;
+    if (!Object.keys(emailFrequencyEnum).includes(selectedFrequency)) {
+      console.error('Invalid email frequency selected:', selectedFrequency);
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/EmailFrequency.jsx:40]: please use setMessage

File: client-side/src/components/settings/manageNotifications/Settings.jsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import EmailFrequency from './EmailFrequency.jsx';
+import TimeZone from './TimeZone.jsx'
+
+const Settings = () => {
+  return (
+    <div className="manage-notifications">
+      <h2>Settings</h2>
+      <EmailFrequency></EmailFrequency>
+      <TimeZone></TimeZone>
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/Settings.jsx]: please remove timeZone

File: client-side/src/components/settings/manageNotifications/Settings.jsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import EmailFrequency from './EmailFrequency.jsx';
+import TimeZone from './TimeZone.jsx'
+
+const Settings = () => {
+  return (
+    <div className="manage-notifications">
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/Settings.jsx:6]: change manage-notifications to settings, also in folder name

File: client-side/src/components/settings/manageNotifications/TimeZone.jsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
```
Comment (SariGlick) [inline client-side/src/components/settings/manageNotifications/TimeZone.jsx]: remove this file

File: server-side/models/preference.model.js [hunk 11]
```diff
@@ -1,16 +1,10 @@
...
-    soundVoice:{type:String,default:'×××ª ××¢× ×¢×©×'}
-})
-export default mongoose.model('Preferences',PreferenceSchema)
-=======
     emailFrequency:{type:String,default:'never',enum:['never','daily', 'weekly','monthly','yearly']},
-    sendNotificationTime:{type:Number,required:true},
-    soundVoice:{type:String,default:'×××ª ××¢× ×¢×©×'}
+    sendNotificationTime:{type:Number,default:30,required:true},
+    soundVoice:{type:String,default:'×××ª ××¢× ×¢×©×'},
+    timeZone: { type: String,enum:['America/New_York','Europe/London','Asia/Tokyo','Asia/Jerusalem','Europe/Paris','Europe/Madrid'], default: 'Asia/Jerusalem' },
```
Comment (SariGlick) [inline server-side/models/preference.model.js]: remove timezone

File: server-side/controllers/preference.controller.js [hunk 12]
```diff
@@ -1,136 +1,54 @@
...
 
 export const addPreference = async (req, res, next) => {
     try {
-        req.body.soundVoice = req.file.originalname;
-        const newPreferenc = new Preference(req.body);
-        await newPreferenc.save();
-        return res.json(newPreferenc).status(201);
+        if (req.file)
+            req.body.soundVoice = req.file.originalname;
+        const newPreference = new Preference(req.body);
```
Comment (SariGlick) [inline server-side/controllers/preference.controller.js:45]: validate req.body here or in Preference constructor

File: server-side/controllers/profile.controller.js [hunk 13]
```diff
@@ -1,61 +1,81 @@
...
-            res.json(profiles);
-        } catch (err) {
-            res.status(500).json({ message: err.message });
-        }
-    };
 
-    export const createProfile= async (req, res) => {
+export const createProfile = async (req, res, next) => {
+    try {
         const newProfile = new Profiles(req.body);
```
Comment (SariGlick) [inline server-side/controllers/profile.controller.js:21]: validate req.body here or in Profile constructor

File: server-side/controllers/user.controller.js [hunk 14]
```diff
@@ -1,114 +1,89 @@
...
-      name,
-      password: hashedPassword,
-      email,
-    });
+    
+    if (req.file ) {
+     req.body.profileImage=req.file.originalname;
+     req.body.password= await bcrypt.hash(req.body.password, 10);
+
+    const newUser = new Users(req.body);
```
Comment (SariGlick) [inline server-side/controllers/user.controller.js:41]: validate req.body here or in User constructor

File: server-side/controllers/visitedWebSite.controller.js [hunk 15]
```diff
@@ -1,53 +1,66 @@
...
+    try {
+        const visitedWebsites = await VisitedWebsite.find().populate('websiteId').select('-__v');
+        res.json(visitedWebsites);
+    } catch (err) {
+        next({message:err.message,status:500})
+    }
+};
+
+export const createVisitedWebsite = async (req, res,next) => {
+    const newVisitedWebsite = new VisitedWebsite(req.body);
```
Comment (SariGlick) [inline server-side/controllers/visitedWebSite.controller.js:14]: validate req.body here or in VisitedWebsite constructor

File: server-side/controllers/websites.controller.js [hunk 16]
```diff
@@ -28,23 +28,23 @@ export const UpdateWebSite=async(req,res,next)=>{
...
          return res.json(newWebsite).status(201);
     } catch (error) {
-        return next({message:error.message});
+        return next({message:error.message,status:500});
     }
 }
+
 export const addWebSite=async(req,res,next)=>{
     try {
         const newWebsite= new Websites(req.body);
```
Comment (SariGlick) [inline server-side/controllers/websites.controller.js:42]: validate req.body here or in Websites constructor

**Review by SariGlick — CHANGES_REQUESTED** [review]

File: client-side/src/components/settings/manageNotifications/EmailFrequency.jsx (hunk 6 above)
Comment (moriya2004) [reply client-side/src/components/settings/manageNotifications/EmailFrequency.jsx:18]: I didn't understand why it needed to be in state and not get it from a props, so I didn't change it

**Review by SariGlick — APPROVED** [review]

**Pull request merged**